pool.go

```diff
@@ -0,0 +1,166 @@
+package viamrtsp
+
+import (
+	"sync"
+	"time"
+
+	"go.viam.com/rdk/logging"
+)
+
+type poolItem struct {
+	frameWrapper *avFrameWrapper
+	lastAccess   time.Time
+}
+
+type framePool struct {
+	items          []poolItem
+	maxAge         time.Duration
+	preseededCount int
+
+	putCount   int
+	getCount   int
+	cleanCount int
+	newCount   int
+
+	mu           sync.RWMutex
+	stopCleaning chan struct{}
+	logger       logging.Logger
+}
+
+func newFramePool(maxAge, cleanupInterval time.Duration, preseededCount int, logger logging.Logger) *framePool {
+	pool := &framePool{
+		items:          make([]poolItem, 0, preseededCount),
+		maxAge:         maxAge,
+		preseededCount: preseededCount,
+		stopCleaning:   make(chan struct{}),
+		logger:         logger,
+	}
+
+	// Pre-seed the pool
+	for i := 0; i < preseededCount; i++ {
+		pool.items = append(pool.items, poolItem{
+			frameWrapper: pool.new(),
+			lastAccess:   time.Now(),
+		})
+	}
+
+	go pool.cleanupRoutine(cleanupInterval)
+
+	return pool
+}
+
+func (p *framePool) new() *avFrameWrapper {
+	p.logger.Debug("newFunc for pool was called!")
+	avFrame, err := newAVFrameWrapper()
+	if err != nil {
+		p.logger.Errorf("Failed to allocate AVFrame in frame pool: %v", err)
+		return nil
+	}
+	p.newCount++
+	return avFrame
+}
+
+func (p *framePool) get() *avFrameWrapper {
+	p.mu.RLock()
```

Comment:
The slicing on lines 71/72 is not safe to be done with 2 concurrent callers to `get`. This needs to be a "write lock". I expect the reader lock "works" here because `get` is only called from a single thread.

Given this is the only use of the "read"ing lock, just changing the `sync.RWMutex` to use a `sync.Mutex` is the right change to make.

Reply:
Done